pkg/detector/detector.go

```diff
@@ -323,7 +323,25 @@ func (d *ResourceDetector) OnAdd(obj interface{}) {
 	if !ok {
 		return
 	}
-	d.Processor.Enqueue(ResourceItem{Obj: runtimeObj})
+	unstructuredObj, err := helper.ToUnstructured(obj)
+	if err != nil {
+		klog.Errorf("Failed to transform newObj, error: %v", err)
+		return
+	}
+	var resourceChangeByKarmada bool
+	switch {
+	case unstructuredObj.GetAnnotations()[policyv1alpha1.PropagationPolicyNameLabel] != "":
+		resourceChangeByKarmada = false
+	case unstructuredObj.GetAnnotations()[policyv1alpha1.ClusterPropagationPolicyLabel] != "":
+		resourceChangeByKarmada = false
+	case unstructuredObj.GetLabels()[policyv1alpha1.PropagationPolicyNameLabel] != "":
+		resourceChangeByKarmada = false
+	case unstructuredObj.GetLabels()[policyv1alpha1.ClusterPropagationPolicyLabel] != "":
+		resourceChangeByKarmada = false
+	default:
+		resourceChangeByKarmada = true
```

Comment:
Supposing user create a resource, in most case the object doesn't have `policy label`, so the `switch` will enter default case.

Then you mark `resourceChangeByKarmada = true`, this will lead to creation the this object doesn't execute `CreateOrUpdateResourceBinding` when the matched policy is `Lazy`, this is not expected.